src/test/java/pages/CataloguePage.java

```diff
@@ -0,0 +1,75 @@
+package pages;
+
+import elements.Text;
+import org.openqa.selenium.By;
+import org.openqa.selenium.TimeoutException;
+import org.openqa.selenium.support.FindBy;
+import org.testng.Assert;
+import util.FindElementUtils;
+import util.WaitUtils;
+
+
+public class CataloguePage extends BasePage {
+
+    private static final String iconOptions = "//li[@data-id=%d]";
+    private static final String leftDropdown = "//div[@data-id=%d]/descendant::div[4]";
+    private static final String middleDropdown = "//div[@data-id=%d]/descendant::span[1]";
```

Comment:
How about implement a method 'format' inside element's class so that you could you '%s' inside @findby and call the element like "leftDropdownElement.format("NeededData").click()"

Reply:
Please remember that number of parameters to format can vаry